geonode_mapstore_client/client/js/components/Accordion/Accordion.jsx

```diff
@@ -46,24 +60,38 @@ const Accordion = ({
     content,
     loadItems,
     items,
-    query
+    query,
+    defaultExpanded,
+    expanded: keepExpanded
 }) => {
     const isMounted = useIsMounted();
+    const clickRef = useRef();
 
     const [accordionsExpanded, setAccordionsExpanded] = useLocalStorage('accordionsExpanded', []);
     const [accordionItems, setAccordionItems] = useState(items);
     const [loading, setLoading] = useState(false);
 
+    useEffect(() => {
+        if ((defaultExpanded || keepExpanded)
+            && !clickRef.current
+            && (isEmpty(accordionsExpanded) || !accordionsExpanded.includes(identifier))
+        ) {
+            isMounted(() => setAccordionsExpanded(uniq(accordionsExpanded.concat(identifier))));
+        }
+    }, [defaultExpanded, keepExpanded, accordionItems, accordionsExpanded]);
+
```

Comment:
I think this side effect make the code too complex for what we are expecting. The behaviour of accordion has been designed to keep the state of the expanded properties while navigating. If we want to include a default expanded property I think we should just apply a default value to `useLocalStorage('key', defaultValue)` or improve useLocalStorage to expect default value function to modify the current saved state `useLocalStorage('key', (defaultValue) => defaultValue)`. I think we should change approach for this because what we are expecting is:

- filter a specific facet config for identifier
- render a list of item with the title without

The filter `group` type in combination with facet could be the UI that we are looking for. This mean we need an additional `group` type for filter facet, currently are available only `select` and `accordion`

```js
// new filter type
{
    type: "group",
    facet: "thesaurus",
    name: ['my-facet']
},
```
In case of facet of type `thesaurus` we could have multiple facet so we could use the `name` property to filter only the one we want to display. I'm proposing `name` bacause it is the same property we have on the endpoint payload. If name is not available all the facets will be displayed. If possible the name option should be implement for all type of facet.

The items in the group type could have the same structure of the accordion children with the possibility to render them with the `style` property as `facet` or `checkbox`. If image or icon is available it should be rendered before the text and after the checkbox if present

Reply:
instead of name we will use include/exclude

```js
{
    type: "group", // select or accordion
    facet: "thesaurus",
    include: ['my-facet'] // include only facet with this names
},
```

```js
{
    type: "group", // select or accordion
    facet: "thesaurus",
    exclude: ['my-facet'] // exclude only facet with this names
},
```

Reply:
@allyoucanmap looks good. A couple of questions:

- How will the one or multiple thesaurus be rendered? I guess every thesaurus will have its own title/section, right?
- In case `include`/`exclude` are not set all the available thesaurus will be rendered, right

Reply:
> How will the one or multiple thesaurus be rendered? I guess every thesaurus will have its own title/section, right?

The will behave like now, if you have multiple thesaurus each one has its own select, accordion or group based on confuguration, include and exclude just pick one or more of them

> In case include/exclude are not set all the available thesaurus will be rendered, right

yes, it will loop through all available thesaurus